code/game/objects/structures/girders.dm

```diff
@@ -32,13 +32,22 @@
 	if(reinf_material)
 		LAZYAPLUS(., reinf_material.name, 2)
 
-/obj/structure/girder/attack_generic(var/mob/user, var/damage, var/attack_message = "smashes apart", var/wallbreaker)
-	if(!damage || !wallbreaker)
-		return 0
-	user.do_attack_animation(src)
-	visible_message(SPAN_DANGER("[user] [attack_message] the [src]!"))
-	spawn(1) dismantle(user)
-	return 1
+/obj/structure/girder/attack_generic(mob/M, damage, attack_message = "smashes apart")
+	log_and_message_admins("attacked [src]: [jumplink(src)]")
+	M.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
+	if(!damage)
+		return attack_hand(M)
+
+	if(damage < 20)
+		playsound(loc, 'sound/effects/metalhit2.ogg', 50, 1)
+		M.do_attack_animation(src)
+		M.visible_message(SPAN_DANGER("\The [M] [attack_message] \the [src]!"))
+		return take_damage(damage*2.5)
+	if(damage >= 30)
+		playsound(loc, 'sound/effects/metal_crash.ogg', 50, 1)
+		M.do_attack_animation(src)
+		M.visible_message(SPAN_DANGER("\The [M] smashes through \the [src]!"))
+		dismantle()
```

Comment:
```suggestion
/obj/structure/girder/attack_generic(mob/M, damage, attack_message = "smashes apart")
	if(damage)
		M.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
		M.do_attack_animation(src)
		M.visible_message(SPAN_DANGER("\The [M] [attack_message] \the [src]!"))
		playsound(loc, 'sound/effects/metalhit2.ogg', 50, 1)
		take_damage(damage)
	else
		attack_hand(M)
```
What are those, cardboard? Smh.

Reply:
Well.. this'd make girders tougher then the actual wall, which doesn't feel right. I agree the instant dismantle is OP but I think mobs should have some multiplier on girder attacks.

Reply:
> Well.. this'd make girders tougher then the actual wall, which doesn't feel right. I agree the instant dismantle is OP but I think mobs should have some multiplier on girder attacks.

Just reduce HP if it's too tough.

Reply:
Better yet, don't. Girder is basically still a wall that you can see and shoot trough.

Reply:
Girders have 150 HP normally and reinforced girders have 500 health, reducing their HP will affect balance unrelated to mobs smashing walls though.

Reply:
Then leave it alone. For wall breaking mekas have drills and related special interactions.

Reply:
Afaik drills are the only tools mechs can use to bypass a wall/girder besides shooting it and unlike hand weapons mechs/mobs don't have structure damage multipliers so a 30 damage hammer (3* multiplier) will do 90 damage when a mech will do 45-60 damage.